Git Formats/tests/syntax_test_git_diff.patch

```diff
@@ -0,0 +1,178 @@
+
+-- >8 ----- >8 ----
+\^^^^^^^^^^^^^^^^^^ meta.block.git-header.diff punctuation.section.block.git
+\  ^^ support.function.scissors.git
+\           ^^ support.function.scissors.git
+Subject: [RFC PATCH v3] How about this instead?
+\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.block.git-header.diff meta.mapping.diff
+\         ^^^ storage.modifier.diff
+\                   ^^ meta.annotation.patch-version.diff storage.modifier.diff
+\^^^^^^ keyword
+\      ^ punctuation.separator.mapping.key-value.diff
+\        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ markup.heading.diff
+\        ^^^^^^^^^^^^^^ variable.annotation.diff
+\        ^ punctuation.definition.annotation.begin.diff
+\                     ^ punctuation.definition.annotation.end.diff
+
+> Do the same for ia64 so we can have sleek & trim looking
+\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.block.git-body.diff markup.quote.git
+\ <- punctuation.definition.blockquote.git
+
+Much better, no?
+\^^^^^^^^^^^^^^^^ meta.block.git-body.diff
+
+---
+ Gemfile                      |   1 +
+\^^^^^^^ entity.name.diff
+\                             ^ punctuation.separator.sequence.diff
+\                                 ^ meta.number.integer.decimal.diff constant.numeric.value.diff
+\                                   ^ markup.inserted.diff
+ app/assets/images/logo.jpeg  | Bin 0 -> 50966 bytes
+\^^^^^^^^^^^^^^^^^^^^^^^^^^^ entity.name.diff
+\   ^ punctuation.separator.sequence.diff
+\          ^ punctuation.separator.sequence.diff
+\                 ^ punctuation.separator.sequence.diff
+\                             ^ punctuation.separator.sequence.diff
+\                               ^^^ storage.type.diff
+\                                   ^ meta.number.integer.decimal.diff constant.numeric.value.diff markup.deleted.diff
+\                                     ^^ keyword.operator.logical.diff
+\                                        ^^^^^ meta.number.integer.decimal.diff constant.numeric.value.diff markup.inserted.diff
+\                                              ^^^^^ constant.numeric.suffix.diff
+ fonts/icomoon.eot            | Bin 55452 -> 56000 bytes
+ 3 files changed, 1 insertion(+)
+\^ meta.number.integer.decimal.diff constant.numeric.value.diff
+\               ^ punctuation.separator.sequence.diff
+\                 ^ meta.number.integer.decimal.diff constant.numeric.value.diff
+\                   ^^^^^^^^^^^^ markup.inserted.diff
+\                            ^ punctuation.definition.diff
+\                              ^ punctuation.definition.diff
+
+diff --git a/Gemfile b/Gemfile
+index c661619..989efe8 100644
+--- a/Gemfile
++++ b/Gemfile
+@@ -24,6 +24,7 @@ group :development do
+   gem 'listen', '~> 3.0.5'
+   gem 'spring'
+   gem 'spring-watcher-listen', '~> 2.0.0'
++  gem 'rspec-rails'
+ end
+
+ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+diff --git a/app/assets/images/logo.jpeg b/app/assets/images/logo.jpeg
```

Comment:
In current Git Log format, lines like these are marked as `source.shell`. Should I do that here, too, or is this way fine?

Reply:
The primary reason I think is re-using _Bash_ syntax to highlight this line.

I don't think we need `source.shell`, but maybe a function-call like scope may be usefull, so ST displays those `diff` lines with proper kind icon in symbol list. Currently it displays `v` (variable?).

Reply:
Added a proposal in 15622fc including updated interpretation of argument scopes. I am uncertain whether `entity.name` is a proper scope for path names though. I'd tend to just scope them string. However saw `entity.name` also being used above and wonder whether all paths should receive a common scope.

Reply:
I think I borrowed scope names for paths from Git Ignore. I'm fine with a scope change, so long as things are consistent.

Reply:
Whatever the scope is, we should make sure at least the paths (ideally together) are available in the local symbols. "Together" so that you don't have to scroll through each one separately in the symbol list. They're almost always the same path anyway.

Reply:
Symbol list should show the whole line due to everything being in `meta.toc-list`.

Reply:
Ah, good. I was on phone when I wrote that above.

What scope do you like? `meta.path string.unquoted`?

Reply:
This was what came to my mind first as it would match closest with current shell argument scopes, which use `string`. I haven't however double checked all locations of possible file names and paths and whether `string` would be appropriate there.

Finally those syntaxes are so special purpose, we don't need to turn scope naming into rocket science.